src/animation/PropertyBinding.js

```diff
@@ -102,47 +102,81 @@ Object.assign( PropertyBinding, {
 
 	},
 
-	parseTrackName: function ( trackName ) {
+	parseTrackName: function () {
 
-		// matches strings in the form of:
-		//    nodeName.property
-		//    nodeName.property[accessor]
-		//    nodeName.material.property[accessor]
-		//    uuid.property[accessor]
-		//    uuid.objectName[objectIndex].propertyName[propertyIndex]
-		//    parentName/nodeName.property
-		//    parentName/parentName/nodeName.property[index]
-		//    .bone[Armature.DEF_cog].position
-		//    scene:helium_balloon_model:helium_balloon_model.position
-		// created and tested via https://regex101.com/#javascript
+		// Parent directories, delimited by '/' or ':'. Currently unused, but must
+		// be matched to parse the rest of the track name.
+		var directoryRe = /((?:[\w-]+[\/:])*)/;
 
-		var re = /^((?:[\w-]+[\/:])*)([\w-]+)?(?:\.([\w-]+)(?:\[(.+)\])?)?\.([\w-]+)(?:\[(.+)\])?$/;
-		var matches = re.exec( trackName );
+		// Target node. May contain word characters (a-zA-Z0-9_) and '.' or '-'
+		// characters, but must begin and end with a word character.
+		var nodeRe = /(\w(?:[\w-\.]*\w)?)?/;
```

Comment:
> but must begin and end with a word character
Why is this limitation necessary?

`nodeName` is compared to `.name` which is general strings so could begin and end with non-word character like `-something-`

https://github.com/donmccurdy/three.js/blob/d6bf94269463418204213dd4f7efdc417311cc24/src/animation/PropertyBinding.js#L198
https://github.com/donmccurdy/three.js/blob/d6bf94269463418204213dd4f7efdc417311cc24/src/animation/PropertyBinding.js#L229